src/libraries/Common/src/Interop/Linux/cgroups/Interop.cgroups.cs

```diff
@@ -333,23 +333,23 @@ internal static bool TryFindCGroupPathForSubsystem(CGroupVersion cgroupVersion,
                                 // list. See man page for cgroups for /proc/[pid]/cgroups format, e.g:
                                 //     hierarchy-ID:controller-list:cgroup-path
                                 //     5:cpuacct,cpu,cpuset:/daemons
-                                if (Array.IndexOf(lineParts[1].Split(','), subsystem) < 0)
+                                if (Array.IndexOf(line[lineParts[1]].Split(','), subsystem) < 0)
                                 {
                                     // Not the relevant entry.
                                     continue;
                                 }
 
-                                path = lineParts[2];
+                                path = line[lineParts[2]];
                                 return true;
                             }
                             else if (cgroupVersion == CGroupVersion.CGroup2)
                             {
                                 // cgroup v2: Find the first entry that matches the cgroup v2 hierarchy:
                                 //     0::$PATH
 
-                                if ((lineParts[0] == "0") && (lineParts[1].Length == 0))
+                                if ((lineSpan[lineParts[0]].SequenceEqual("0")) && (lineSpan[lineParts[1]].Length == 0))
```

Comment:
```suggestion
                                if ((lineSpan[lineParts[0]] is "0") && (lineSpan[lineParts[1]].Length == 0))
```
Can be sugared using [Pattern matching on ROS](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/pattern-match-span-of-char-on-string.md)